### DISCUSSION TITLE
Groth16 Plonk zkp prove supported[Feature Request] <Title>

### DISCUSSION THREAD

**Discussion by linyoufa** [zkSync-Community-Hub/zksync-developers#209] [Feature requests]
### Network

zkSync Era

### Provide a brief description of the feature you would like to see implemented.

🌟 Feature Request

Operations such as bn256 pairing multiplication do not currently have a corresponding precompiled function, but are supported in eth
📝 Description

Provide a clear and concise description of the feature you'd like to see.
1.The following method calls failed

success := staticcall(sub(gas(), 2000000), 6, mIn, 128, pR, 64)
success := staticcall(sub(gas(), 2000), 7, mIn, 96, mIn, 64)
success := staticcall(sub(gas(), 2000), 8, _pPairing, 768, _pPairing, 0x20)

2.this is pre-compiled Ethereum:

var PrecompiledContractsIstanbul = map[common.Address]PrecompiledContract{
common.BytesToAddress([]byte{1}): &ecrecover{},
common.BytesToAddress([]byte{2}): &sha256hash{},
common.BytesToAddress([]byte{3}): &ripemd160hash{},
common.BytesToAddress([]byte{4}): &dataCopy{},
common.BytesToAddress([]byte{5}): &bigModExp{eip2565: false},
common.BytesToAddress([]byte{6}): &bn256AddIstanbul{},
common.BytesToAddress([]byte{7}): &bn256ScalarMulIstanbul{},
common.BytesToAddress([]byte{8}): &bn256PairingIstanbul{},
common.BytesToAddress([]byte{9}): &blake2F{},
}

3.zksync precompiles
Ecrecover.yul Keccake256.yul SHA256.yul
🤔 Rationale

Explain why this feature is important and how it benefits the project.

After using zk-snark algorithm to generate zkp proof, users need to perform zkp prove in zksync
📋 Additional Context

Add any other context or information about the feature request here.
my email: [linyoufa@gmail.com](mailto:linyoufa@gmail.com)

### Who does this feature request benefit the most (users, developers, protocol, etc.)?

users  and developers

**Comment by rietracerdanchri** [comment]
This feature request seeks to implement precompiled contracts for operations like bn256 pairing multiplication in zkSync Era. This enhancement would benefit users and developers by enabling efficient execution of zk-snark algorithms and zkp proofs within zkSync.

**Comment by neumagbutorcapri** [comment]
Implement precompiled functions for bn256 pairing multiplication in zkSync Era to enable efficient zk-SNARK proof verification, benefiting both users and developers.

**Comment by singletonbrittany** [comment]
Feature Request: Implement precompiled functions for operations like bn256 pairing multiplication in zkSync Era to enable efficient zk-proof verification. This enhancement benefits both users and developers by streamlining the process of zk-proof validation. #zkSync #zkProof #EraImprovement.

**Comment by swelexdewi** [comment]
The user is requesting the implementation of a precompiled function for operations such as bn256 pairing multiplication in the zkSync Era network. They have provided code examples and rationale for the importance of this feature in performing zero-knowledge proof in zkSync. The request would benefit both users and developers. The user has also provided their email for further communication.

**Comment by duongnd-183902** [comment]
still wait :D
